Cargo.toml

```diff
@@ -14,10 +14,17 @@ edition = "2018"
 parking_lot_core = { path = "core", version = "0.5" }
 lock_api = { path = "lock_api", version = "0.2" }
 
+# Optional dependency for supporting serde.  Use the feature "enable_serde" to
+# get access to this.
+serde = {version = "1.0.90", features = ["derive", "rc", "std"], optional = true}
```

Comment:
`parking_lot` doesn't need to depend on `serde`. Only `lock_api` needs the dependency.

Reply:
OK.  Do you want me to make all these changes, or do you want to do it and merge them back in?

Reply:
Yes, but if you don't know that, then you end up in the situation I was in with my project where all the tests from within the `lock_api` crate work, but once you pop up outside of it they don't because `parking_lot` isn't enabling `serde`.  By putting the optional feature into the top-level crate (`parking_lot`) new users are aware of the existence of the `enable_serde` feature, and can use it directly.

In addition, by putting `enable_serde` into `parking_lot`, you can turn it on and off on all of the dependent crates at the same time; this could be useful if you need to enable it for the `core` or `benchmark` crates at the same time.

Reply:
You can add a "proxy" serde feature to parking_lot, which just forwards to lock_api/serde:

```
[features]
serde = ["lock_api/serde"]
```

Reply:
OK, I've added the proxy serde feature.  That said, I'm still using `enable_serde` and not `serde`.  Should I change that as well?  I think I remember seeing a comment online that if the name of a crate and a feature switch are the same, then there will be issues with Cargo, but I don't remember where I saw that... is that correct?

Reply:
You should replace `enable_serde` with just `serde`. You don't need to add a `serde` feature to `lock_api` since optional crates will automatically create a feature with the same name as the crate.

Reply:
OK, I made the changes.  Thank you for explaining that about Cargo to me!